Sources/PaymentsCore/Networking/APIClient.swift

```diff
@@ -1,84 +1,51 @@
+        if let urlRequest = urlRequest(from: apiRequest) {
+            http.send(urlRequest, completion: completion)
+        } else {
+            // TODO: handle error
         }
+    }
 
-        urlSession.performRequest(with: request) { data, response, error in
-            let correlationID = (response as? HTTPURLResponse)?.allHeaderFields["Paypal-Debug-Id"] as? String
-
-            if let error = error {
-                completion(.failure(.connectionIssue(error)), correlationID)
-                return
-            }
-
-            guard let response = response as? HTTPURLResponse else {
-                completion(.failure(.invalidURLResponse), correlationID)
-                return
-            }
+    private func urlRequest(from apiRequest: APIRequest) -> URLRequest? {
+        let completeUrl = config.environment.baseURL.appendingPathComponent(apiRequest.path)
+        let urlComponents = URLComponents(url: completeUrl, resolvingAgainstBaseURL: false)
 
-            guard let data = data else {
-                completion(.failure(.noResponseData), correlationID)
-                return
-            }
+        guard let url = urlComponents?.url else {
+            return nil
+        }
 
-            switch response.statusCode {
-            case 200..<300:
-                do {
-                    let decodedData = try self.decoder.decode(T.ResponseType.self, from: data)
-                    completion(.success(decodedData), correlationID)
-                    return
-                } catch {
-                    // TODO: Returning this error will always be nil at this point
-                    completion(.failure(.parsingError(error)), correlationID)
-                    return
-                }
+        var request = URLRequest(url: url)
+        request.httpMethod = apiRequest.method.rawValue
+        request.httpBody = apiRequest.body?.encoded()
 
-            default:
-                // TODO:
-                // Add networking error cases (ie more descriptive networking errors / handle 400 responses, 500 errors, etc
-                completion(.failure(.unknown), nil)
-                return
-            }
-        }
-    }
-}
+        // TODO: Remove clientSecret when this endpoint is updated and can be used with low-scoped token
+        // For now, include your clientSecret here and your clientID in CoreConfig to test this request
+        let clientSecret = ""
+        let encodedClientID = "\(config.clientID):\(clientSecret)".data(using: .utf8)?.base64EncodedString() ?? ""
 
-protocol URLSessionProtocol {
-    func performRequest(with urlRequest: URLRequest, completionHandler: @escaping (Data?, URLResponse?, Error?) -> Void)
-}
+        request.addValue("application/json", forHTTPHeaderField: "Content-Type")
+        request.addValue("en_US", forHTTPHeaderField: "Accept-Language")
+        request.addValue("Basic \(encodedClientID)", forHTTPHeaderField: "Authorization")
```

Comment:
The `APIClient` is `CoreConfig` aware, and it is capable of turning an `APIRequest` into a `URLRequest`.

Reply:
Then again - I haven't worked with them before. Curious what you and @jonathajones think on that?

Reply:
I would say that protocol extensions aren't unusual - but they should be used sparingly, since they can lead to unusual behavior depending on how the concrete object is being cast when using a protocol function with a default implementation.

Our previous implementation uses a protocol to describe a url request, and i think that's still the right way to go. There isn't a reason to use a reference type (class / subsclass) to describe a network request - an api request should only really hold the data needed to to help the network layer create a network request, which can be done by a value type more safely (we won't have to worry about mutability or shared access to the reference type). Since structs can't inherit from classes, i think that using a protocol interface is still preferable to subclassing / inheritance. 

practically, since this is not something that we will expose to merchants, either approach would work - but i do think using a protocol interface with conforming structs is a safer way to get the job done. 

The protocol extension itself was a way of pulling shared logic out of the individual types, but still provide a way for the concrete types to create the url requests a different way (by implementing the `toURLRequest(environment:)` protocol function on the conforming struct

Reply:
Ah makes sense! Thanks for clarifying that.